plugin/core/types.py

```diff
@@ -683,10 +685,14 @@ def map_client_path_to_server_uri(self, path: str) -> str:
                 path, mapped = path_map.map_from_local_to_remote(path)
                 if mapped:
                     break
-        return filename_to_uri(path)
+        return urljoin('file:', pathname2url(path))
```

Comment:
I don't understand why we are duplicating the code here if we have to keep original functions anyway. There is a chance for implementations to get out of sync that way.

Reply:
Because filename_to_uri and uri_to_filename would ideally be removed, but we cannot do that because they're part of the public api.

Once this is merged I'll go ahead and change all the functions in plugin/core/views.py to take a ClientConfig, and those functions will then call filename_to_uri and uri_to_filename with an actual ClientConfig. At the same time LSP-* packages have some time to upgrade.

Reply:
I think those could already call filename_to_uri and uri_to_filename but I trust you will clean it up later. :)